data_structures/disjoint_set/disjoint_set.py

```diff
@@ -0,0 +1,62 @@
+class node:
+    def __init__(self, data):
+        self.data = data
+
+
+def makeSet(x):
+    """
+    make x as a set.
+    """
+    # rank is the distance from x to its' parent
+    # root's rank is 0
+    x.rank = 0
+    # p is x's parent
+    x.p = x
+
+
+def unionSet(x, y):
+    """
+    union two sets.
+    set with bigger rank should be parent, so that the
+    disjoint set tree will be more flat.
+    """
+    x, y = findSet(x), findSet(y)
+    if x.rank > y.rank:
+        y.p = x
+    else:
+        x.p = y
+        if x.rank == y.rank:
+            y.rank += 1
+
+
+def findSet(x):
+    """
+    return the parent of x
+    """
+    if x != x.p:
+        x.p = findSet(x.p)
+    return x.p
+
+
+def test_disjoint_set():

```

Comment:
Please add a doctest here so that our automated tests will pick this up.
```python
def test_disjoint_set():
    """
    >>> test_disjoint_set()
    """